webapp/src/ts/services/xml-forms.service.ts

```diff
@@ -268,7 +279,10 @@ export class XmlFormsService {
       })
       .then(doc => {
         if (!this.findXFormAttachmentName(doc)) {
-          return Promise.reject(new Error(`The form "${internalId}" doesn't have an xform attachment`));
+          const errorMessage = `The form "${internalId}" doesn't have an xform attachment`;
+          this.feedbackService.submit('Error in XMLFormService findXFormAttachmentName : ' + errorMessage, false);
+          console.error('Error in XMLFormService findXFormAttachmentName : ', errorMessage);
+          return Promise.reject(new Error(errorMessage));
         }
         return doc;
       });
```

Comment:
So we don't need multiple calls to `feedbackService.submit`, can you please add a catch all here that will record feedback, and remove the 3 other ones: 
```
.catch(err => {
  this.feedbackService.submit('Error in XMLFormService : ' + err.message, false);
  throw err;
});
```